synapse/handlers/federation_event.py

```diff
@@ -1809,7 +1809,24 @@ async def _run_push_actions_and_persist_event(
 
         try:
             await self.persist_events_and_notify(
-                event.room_id, [(event, context)], backfilled=backfilled
+                event.room_id,
+                [(event, context)],
+                # We should not send notifications about backfilled events.
+                inhibit_push_notifications=backfilled,
+                # We don't need to calculate the state for backfilled events and
+                # there is no need to update the forward extrems because we
+                # already know this event happened in the past if it was
+                # backfilled.
+                should_calculate_state_and_forward_extrems=not backfilled,
+                # Backfilled events get a negative stream ordering so they don't
+                # come down incremental `/sync`
+                use_negative_stream_ordering=backfilled,
+                # Backfilled events do not affect the current local state
+                inhibit_local_membership_updates=backfilled,
```

Comment:
would it be sensible to have this implied by `should_calculate_state_and_forward_extrems` ?

Reply:
I understand the leap that updating the state in the room also encompasses member state events.

My first impression was that it wasn't very obvious but it makes sense ⏩